cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -44,23 +44,30 @@
 namespace cudf {
 
 /**
- * @brief Policy for what assumptions the optional iterator has about null values
+ * @brief Indicates the presence of nulls at compile-time or runtime.
  *
- * - `YES` means that the column supports nulls and has null values, therefore
- *    the optional might not contain a value
+ * If used at compile-time, this indicator can tell the optimizer
+ * to include or exclude any null-checking clauses.
  *
- * - `NO` means that the column has no null values, therefore the optional will
- *    always have a value
- *
- * - `DYNAMIC` defers the assumption of nullability to runtime with the users stating
- *    on construction of the iterator if column has nulls.
  */
-struct contains_nulls {
-  struct YES {
+struct nullate {
```

Comment:
This should be in `detail::`

Reply:
Are you sure? It is used for parameters on `cudf` namespace'd functions and classes. Seems odd to require a `detail` parameter on a non-detail function/class.

Reply:
Really? I wouldn't think any public functionality would use this given that functionality would necessarily be a template.

Reply:
The main usage (in this PR) is by the row-comparators which are all in `cudf` namespace. It is also used as a parameter to the `optional_begin()` and `optional_end()` methods on the `cudf::column_device_view` class.

Reply:
Hm, I guess that's true. Okay, we can leave it for now.